vscode-ext/src/extension.ts

```diff
@@ -24,90 +48,33 @@ function getK8sApi(): CoreV1Api {
-		if (setting === undefined) {
-			return;
-		}
-		else if (setting.startsWith('Toggle telemetry')) {
-			globalContext.workspaceState.update('telemetry', !telemetry);
-		}
-		else if (setting.startsWith('Toggle file')) {
-			globalContext.workspaceState.update('fileOps', !fileOps);
-		}
-		else if (setting.startsWith('Toggle invalid certificates')) {
-			globalContext.workspaceState.update('invalidCertificates', !invalidCertificates);
-		}
-		else if (setting.startsWith('Toggle traffic stealing')) {
-			globalContext.workspaceState.update('trafficStealing', !trafficStealing);
-		}
-		else if (setting.startsWith('Toggle remote DNS')) {
-			globalContext.workspaceState.update('remoteDNS', !remoteDNS);
-		}
-		else if (setting.startsWith('Toggle outgoing traffic')) {
-			globalContext.workspaceState.update('outgoingTraffic', !outgoingTraffic);
-		}
-		else if (setting.startsWith('Include environment variables')) {
-			vscode.window.showInputBox({ prompt: 'Enter environment variables to include (separated by commas, * for all)', value: includeEnvironmentVariables }).then(async value => {
-				if (value === undefined) {
-					return;
-				}
-				globalContext.workspaceState.update('includeEnvironmentVariables', value);
-			});
-		}
-		else if (setting.startsWith('Exclude environment variables')) {
-			vscode.window.showInputBox({ prompt: 'Enter environment variables to exclude', value: excludeEnvironmentVariables }).then(async value => {
-				if (value === undefined) {
-					return;
-				}
-				globalContext.workspaceState.update('excludeEnvironmentVariables', value);
-			});
-		} else if (setting.startsWith('Change namespace')) {
-			const namespaces: {
-				response: any;
-				body: V1NamespaceList;
-			} = await getK8sApi().listNamespace();
-			const namespaceNames = namespaces.body.items.map(namespace => namespace.metadata!.name!);
-			vscode.window.showQuickPick(namespaceNames, { placeHolder: 'Select namespace' }).then(async namespaceName => {
-				if (namespaceName === undefined) {
-					return;
-				}
-				if (setting.startsWith('Change namespace for mirrord agent')) {
-					globalContext.workspaceState.update('agentNamespace', namespaceName);
-				} else if (setting.startsWith('Change namespace for impersonated pod')) {
-					globalContext.workspaceState.update('impersonatedPodNamespace', namespaceName);
-				}
-			});
-		}
-	});
+async function configFilePath() {
+	let vsCodeDir = VS_CODE_DIR;
+	let fileUri = vscode.Uri.joinPath(vsCodeDir, 'mirrord.+(toml|json|y?(a)ml)');
+	let files = glob.sync(fileUri.fsPath);
+	if (files.length !== 0) {
+		let file = files[0];
+		return file;
+	}
+	else {
+		return '';
+	}
 }
 
+async function openConfig() {
+	let path = await configFilePath();
+	if (!path) {
+		path = vscode.Uri.joinPath(VS_CODE_DIR, 'mirrord.json');
+		await vscode.workspace.fs.writeFile(path, Buffer.from(DEFAULT_CONFIG));
+	}
+	vscode.workspace.openTextDocument(path).then(doc => {
+		vscode.window.showTextDocument(doc);
+	});
```

Comment:
a .catch() would be nice

Reply:
When would it be called?

Reply:
```
vscode.workspace.openTextDocument(path).then(doc => {
		vscode.window.showTextDocument(doc);
	}).catch(err: Any => {
// foo
});
```
just meant that, similar to my comment before maybe we can handle any exceptions and let the user know?

Reply:
A. Let the user know what, though? If we don't know possible causes for this exception, then what can we write that would be more helpful than what's being written automatically?
B. Since we can't come up with any actual scenario where this would throw an exception, then I see no reason to clutter the codebase with a catch clause that would never be called. I prefer to take the risk that there's something I haven't thought about (this reasoning doesn't apply to the mirrord-agent/layer by the way).

Reply:
okay, sounds good